postproengine/wake_meandering.py

```diff
@@ -0,0 +1,365 @@
+        actionname = 'statistics'
+        blurb      = 'Compute wake meandering statistics'
+        required   = False
+        actiondefs = [
+        {'key':'savefile',  'required':False,  'default':'',
+         'help':'Filename to save statistics', },
+        {'key':'mean',  'required':False,  'default':True,
+         'help':'Boolean to compute mean wake center', },
+        {'key':'std',  'required':False,  'default':True,
+         'help':'Boolean to compute std wake center', },
+        {'key':'anisotropy',  'required':False,  'default':False,
+         'help':'Boolean to compute wake anisotropy metric', },
+        ]
+        
+        def __init__(self, parent, inputs):
+            self.actiondict = mergedicts(inputs, self.actiondefs)
+            self.parent = parent
+            print('Initialized '+self.actionname+' inside '+parent.name)
+            return
+
+        def execute(self):
+            print('Executing '+self.actionname)
+            savefile = self.actiondict['savefile']
+            computeMean = self.actiondict['mean']
+            computeStd = self.actiondict['std']
+            computeAniso = self.actiondict['anisotropy']
+
+            wake_meandering_stats = {}
+
+            dist_from_hub = np.sqrt( (self.parent.dfcenters['zc']-self.parent.zhub)**2 + (self.parent.dfcenters['yc']-self.parent.yhub)**2 )
+
+            if computeMean:
+                wake_meandering_stats['xc_mean'] = np.mean(self.parent.dfcenters['xc'])
+                wake_meandering_stats['yc_mean'] = np.mean(self.parent.dfcenters['yc'])
+                wake_meandering_stats['zc_mean'] = np.mean(self.parent.dfcenters['zc'])
+                wake_meandering_stats['hub_distance_mean'] = np.mean(dist_from_hub)
+
+            if computeStd:
+                wake_meandering_stats['xc_std'] = np.std(self.parent.dfcenters['xc'])
+                wake_meandering_stats['yc_std'] = np.std(self.parent.dfcenters['yc'])
+                wake_meandering_stats['zc_std'] = np.std(self.parent.dfcenters['zc'])
+                wake_meandering_stats['hub_distance_std'] = np.std(dist_from_hub)
+
+            if computeAniso:
+                try:
+                    numSamples = self.parent.wake.u.shape[0]
+                    eig_ratio = np.zeros(numSamples)
+                    for sample in range(numSamples):
+                        wakeu = self.parent.wake.u[sample,:,:]
+                        wakey = self.parent.wake.y
+                        wakez = self.parent.wake.z
+                        mask = wakeu < self.parent.wake.Clevels[sample]
+                        masku = wakeu * mask
+                        masky = wakey * mask
+                        maskz = wakez * mask
+                        pca_data = np.zeros((len(maskz[maskz != 0]),2))
+                        pca_data[:,0] = masky[maskz != 0]
+                        pca_data[:,1] = maskz[maskz != 0] - np.mean(maskz[maskz != 0])
+                        # if compute_uv:
```

Comment:
Should we make an optional return with the eigenvectors of the 2 leading directions? or just remove this code for now?

Reply:
I'm not sure how useful this anisotropy metric even is, but I've added an option to store the eigenvectors from the principle component analysis in a pklfile

Reply:
I think it's useful to keep this as an optional metric for wake modeling purposes: if nothing else, I will use this ;)